### PR TITLE
Add Javascript implementation of Strategy pattern #67

### PR DESCRIPTION
**Pull request by manasvi111**
In this example:

- I am defining two sets of strategies: authentication strategies (basic authentication and OAuth) and authorization strategies (role-based and permission-based).

- Each strategy encapsulates a specific authentication or authorization logic.

- We define separate contexts (AuthenticationContext and AuthorizationContext) to handle authentication and authorization, respectively.

- Each context can be configured with different strategies at runtime using the setStrategy method.

- Example usage demonstrates how different authentication and authorization strategies can be applied and switched dynamically based on requirements.

### CODE REVIEW

File: patterns/strategy-plan/example.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Define authentication strategies
+const basicAuthStrategy = {
+    authenticate: function(username, password) {
+      // Perform basic authentication logic
+      return username === "admin" && password === "password";
+    }
+  };
+
+const oauthStrategy = {
```
Comment (Dhuds1) [inline patterns/strategy-plan/example.js:10]: What does the `o` in `oauthStrategy` standfor?

File: patterns/strategy-plan/example.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+// Define authorization strategies
+const roleBasedAuthorizationStrategy = {
+    authorize: function(user, role) {
+      // Perform role-based authorization logic
+      return user.role === role;
+    }
+  };
+
+const permissionBasedAuthorizationStrategy = {
```
Comment (Dhuds1) [inline patterns/strategy-plan/example.js:27]: what are we writing in `Java`? 🤣 
this variable name rivals them!

File: patterns/strategy-plan/example.js (hunk 1 above)
Comment (manasvi111) [reply patterns/strategy-plan/example.js:10]: in short: O that means open authorization strategy.
in detail: OAuth (Open Authorization) is an open standard for access delegation commonly used as a way for Internet users to grant websites or applications access to their information on other websites but without giving them the passwords. It allows users to share their resources stored on one site with another site without having to expose their credentials (such as a username and password). OAuth is widely used for enabling third-party applications to access resources on behalf of the user, typically via APIs.

In the context of web development, OAuth is often used for authentication and authorization mechanisms, allowing users to log in to websites or applications using their existing credentials from other services (like Google, Facebook, or GitHub) without sharing their passwords directly with the site.

Hope this helps you to understand!

File: patterns/strategy-plan/example.js (hunk 2 above)
Comment (manasvi111) [reply patterns/strategy-plan/example.js:27]: I am actually writing JavaScript, not Java! JavaScript is a widely used programming language for web development, while Java is a different language often used for other types of applications. As for the variable names, they are chosen according to JavaScript naming conventions. 

Let me know if you have any questions or any other suggestions!

Thank you!

Comment (Dhuds1) [reply patterns/strategy-plan/example.js:27]: Yes I know you're writing in JS 😄 I was making a joke, as Java is know for having very very long variable names!

Comment (manasvi111) [reply patterns/strategy-plan/example.js:27]: Just to avoid confusion to understand what does this use for, I used long names.

Thank you though😄

**Comment by ash-teach** [comment]
This is great work overall, @manasvi111. Could you please include references to any resources used or consulted in the development of this PR?

**Pull request merged**